From 766c081940d609b74c716c0731d638461e05d0af Mon Sep 17 00:00:00 2001
From: Felix Kuehling <Felix.Kuehling@amd.com>
Date: Mon, 18 Apr 2016 18:07:20 -0400
Subject: [PATCH 1520/1722] drm/amd/dal: Fix DM build

The only caller of dm_crtc_pin_cursor_bo_new is interested in the
drm_gem_object. Return that instead of the amdgpu_bo that wraps it.

Change-Id: I69794b8c36c65f7dc3f732479672115d4b84aed9
Signed-off-by: Felix Kuehling <Felix.Kuehling@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index 69d1edb..ec4ebd4 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -142,7 +142,7 @@ static int dm_crtc_pin_cursor_bo_new(
 	struct drm_crtc *crtc,
 	struct drm_file *file_priv,
 	uint32_t handle,
-	struct amdgpu_bo **ret_obj)
+	struct drm_gem_object **ret_obj)
 {
 	struct amdgpu_crtc *amdgpu_crtc;
 	struct amdgpu_bo *robj;
@@ -182,7 +182,7 @@ static int dm_crtc_pin_cursor_bo_new(
 
 		if (ret == 0) {
 			amdgpu_crtc->cursor_addr = gpu_addr;
-			*ret_obj  = robj;
+			*ret_obj  = obj;
 		}
 		amdgpu_bo_unreserve(robj);
 		if (ret)
@@ -201,7 +201,7 @@ static int dm_crtc_cursor_set(
 	uint32_t width,
 	uint32_t height)
 {
-	struct amdgpu_bo *new_cursor_bo;
+	struct drm_gem_object *new_cursor_bo;
 	struct dc_cursor_position position;
 
 	int ret;
@@ -262,7 +262,7 @@ static int dm_crtc_cursor_set(
 	dm_crtc_unpin_cursor_bo_old(amdgpu_crtc);
 
 	/*assign new cursor bo to our internal cache*/
-	amdgpu_crtc->cursor_bo = &new_cursor_bo->gem_base;
+	amdgpu_crtc->cursor_bo = new_cursor_bo;
 
 release:
 	return ret;
-- 
2.7.4

